src/main/java/io/ebeaninternal/server/loadcontext/DLoadBeanContext.java

```diff
@@ -181,14 +180,8 @@ public void loadBean(EntityBeanIntercept ebi) {
 
       if (context.hitCache) {
         // check each of the beans in the batch to see if they are in the L2 cache.
-        Iterator<EntityBeanIntercept> iterator = list.iterator();
-        while (iterator.hasNext()) {
-          EntityBeanIntercept batchEbi = iterator.next();
-          if (batchEbi != ebi && context.desc.cacheBeanLoad(batchEbi, persistenceContext)) {
-            // bean successfully loaded from L2 cache so remove from batch load
-            iterator.remove();
-          }
-        }
+        // bean successfully loaded from L2 cache so remove from batch load
+        list.removeIf(batchEbi -> batchEbi != ebi && context.desc.cacheBeanLoad(batchEbi, persistenceContext)); // Intellij java8 suggestion.
```

Comment:
This is a refactor by Intellij. Java8 has "removeIf" for exactly the given scenario.